src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -1108,6 +1142,25 @@ def compute_several_extensions(self, extensions, save=True, verbose=False, **job
             else:
                 self.compute_one_extension(extension_name, save=save, verbose=verbose, **extension_params)
 
+    def does_extension_need_recompute(self, extension_name, **kwargs):
+        if not self.has_extension(extension_name):
+            return True
+        else:
+            extension_class = get_extension_class(extension_name)
+            if extension_class.need_job_kwargs:
+                params, job_kwargs = split_job_kwargs(kwargs)
+            else:
+                params = kwargs
+
+            extension_instance = extension_class(self)
+            extension_instance.set_params(**params)
+
+            loaded_extension = self.get_extension(extension_name)
+            if loaded_extension.params == extension_instance.params:
```

Comment:
Does this check always work? I didn't realize you could do == for a dict? Or am I misunderstanding this?

Reply:
it seems so :)

Reply:
Yes. One tiny interesting thing. Dictionaries in python are ordered (since 3.6), but their comparison is order-independent. So if
```
a={1:2, 2:4}
b={2:4, 1:2}
```
then
```
a == b
a.keys() == b.keys
```
both return `True`, but 
```
list(a.keys()) == list(b.keys)
```
returns `False`.